app/controllers/application_controller.rb

```diff
@@ -1,2 +1,36 @@
 class ApplicationController < ActionController::API
+
+  before_action :authenticate_user_from_token!
+
+  def authenticate_user_from_token!
+    token = request.headers['Authorization']
+
+    if token
+      authenticate_with_auth_token(token)
+    else
+      authentication_error
+    end
+  end
+
+  def current_user
+    @current_user
+  end
+
+
+  private
+
+  def authenticate_with_auth_token(token)
+    user = User.find_by(access_token: token)
+
+    if user
+      @current_user ||= user
+    else
+      authentication_error
+    end
+  end
+
+  def authentication_error
+    render json: { error: I18n.t('errors.unauthorized') }, status: 401
+  end
+
```

Comment:
This all looks great and will do the job. However, here's a refactoring I'd consider:

```ruby
rescue_from CustomAuthenticationExceptionClassYouName, with: :authentication_error

private

def current_user
  @current_user ||= User.authenticate(auth_token)
end
helper_method :current_user

def auth_token
  request.headers['Authorization']
end

def authentication_error
  render json: { error: I18n.t('errors.unauthorized') }, status: 401
end
```

This moves the logic for where the auth token is to it's own method which makes it easy to change in the future. It moves the authentication logic to the `User` model because that's responsibility the controller shouldn't have. You could also move it to a new object. However, the logic here is simple enough a class method on `User` seems fine. However you move it out of the controller the object should raise a custom error class if the auth fails for any reason. Which we in turn capture in a `rescue_from` statement in the controller.